File: programmers/level1/비밀지도.java
package programmers;

import java.util.Arrays;

public class 비밀지도 {
	static class Solution {
		public String[] solution(int n, int[] arr1, int[] arr2) {
			String[] answer = new String[n];
			for (int i = 0; i < n; ++i) {
				StringBuilder sb = new StringBuilder();
				int temp = arr1[i] | arr2[i];
				for (int j = 0; j < n; ++j) {
					sb.append(temp % 2 == 1 ? '#' : ' ');
					temp /= 2;
				}
				answer[i] = sb.reverse().toString();
			}
			return answer;
		}
	}
	public static void main(String[] args) {
		int n = 5;
		int[] arr1 = { 9, 20, 28, 18, 11 };
		int[] arr2 = { 30, 1, 21, 17, 28 };
		Solution sol = new Solution();
		System.out.println(Arrays.toString(sol.solution(n, arr1, arr2)));
	}
}

File: programmers/python/level1/비밀지도.py
from pprint import pprint
def solution(n, arr1, arr2):
    answer = [[' ' for j in range(n)] for i in range(n)]
    for i, (num1, num2) in enumerate(zip(arr1, arr2)):
        for j in range(n-1, -1, -1):
            if num1 % 2 == 1 or num2 % 2 == 1:
                answer[i][j] = '#'
            num1 //= 2
            num2 //= 2
    answer = [''.join(i) for i in answer]
    return answer

pprint(solution(5, [9, 20, 28, 18, 11], [30, 1, 21, 17, 28]))